m4/slepc.m4

```diff
@@ -51,11 +52,20 @@ AC_DEFUN([CONFIGURE_SLEPC],
           [
             dnl Similar to petsc, we need the slepc version number.
             dnl Note slepc will most likely only work with the corresponding version of petsc
-            slepcmajor=`grep "define SLEPC_VERSION_MAJOR" $SLEPC_DIR/include/slepcversion.h | sed -e "s/#define SLEPC_VERSION_MAJOR[ ]*//g"`
-            slepcminor=`grep "define SLEPC_VERSION_MINOR" $SLEPC_DIR/include/slepcversion.h | sed -e "s/#define SLEPC_VERSION_MINOR[ ]*//g"`
-            slepcsubminor=`grep "define SLEPC_VERSION_SUBMINOR" $SLEPC_DIR/include/slepcversion.h | sed -e "s/#define SLEPC_VERSION_SUBMINOR[ ]*//g"`
-            slepcversion=$slepcmajor.$slepcminor.$slepcsubminor
+            AC_CHECK_HEADER([$SLEPC_DIR/include/slepcversion.h],
```

Comment:
Hmm... calling `AC_CHECK_HEADER` twice on the same file will probably work OK, but this actually triggers a [test compilation](https://www.gnu.org/software/autoconf/manual/autoconf-2.67/html_node/Generic-Headers.html) so it's good to not call it more than is strictly necessary. 

Would you mind setting a flag, e.g. `slepc_in_petsc_arch`, and checking that subsequently instead of calling `AC_CHECK_HEADER` again? Otherwise, this change seems fine to me.

Reply:
Thanks, John. I will do that.

I am thinking should we just check  PETSC_DIR and PETSC_ARCH if slepc is just in the same place as PETSc, and then we can simply set SLEPC_DIR=PETSC_DIR, and then continue configuration.

Reply:
Sounds fine to me. I don't use this type of configuration at all, so as far as I'm concerned, it's up to you exactly how you want to handle it.

Reply:
You mean set SLEPC_DIR=PETSC_DIR/PETSC_ARCH so our later tests are looking in the right place?  That sounds like it would make the code slightly simpler but significantly more confusing, since conceptually SLEPC_DIR is a const global variable.  I like the existing PR better, and I like John's suggestion better still.

But I second "as far as I'm concerned, it's up to you exactly how you want to handle it".  I hate m4 and I'm not proud of our existing m4 and I'm hardly going to hold new m4 to arbitrarily higher standards.

Reply:
> You mean set SLEPC_DIR=PETSC_DIR/PETSC_ARCH so our later tests are looking in the right place? That sounds like it would make the code slightly simpler but significantly more confusing, since conceptually SLEPC_DIR is a const global variable. I like the existing PR better, and I like John's suggestion better still.

What I meant was that if SLEPC_DIR is not set by users, should we try to detect if PETSc carries with SLEPc or not?  In order to accomplish this goal, we could simply set SLEPC_DIR=$PETSC_DIR when users do not set SLEPC_DIR.

Reply:
Oh, sorry, I get it now.  Yeah, I like that.  The more autodetection the better, at least in cases like this where options like ```--disable-slepc``` exist to allow users to override the autodetection.